File: src/LorrisAnalyzer/DataWidgets/ScriptWidget/examples/terminal.py
#  Make ScriptWidget work as terminal
history = [""]
cur_hist = 0
last_idx = -1

def onDataChanged(data, dev, cmd, index):
    if last_idx == index:
        return ""
    terminal.appendText(data)
    last_idx = index
    return ""

def onKeyPress(key):
    lorris.sendData(key)

# Input line - right click to script -> show input line
def interactiveInput():
    global history
    global cur_hist

    lorris.sendData(inputLine.text)

    if cur_hist != len(history)-1:
        history.pop(cur_hist)
    cur_hist = len(history)
    history[cur_hist-1] = inputLine.text;
    history.append("");

    inputLine.clear()

# code is Qt::Key (http://qt-project.org/doc/qt-4.8/qt.html#Key-enum)
def inputLineKeyPressed(code):
    global history
    global cur_hist
    if code == 0x01000013 and cur_hist > 0: # Qt::Key_Up
        cur_hist -= 1
        inputLine.setText(history[cur_hist])
    elif code == 0x01000015 and cur_hist < len(history)-1: # Qt::Key_Down
        cur_hist += 1
        inputLine.setText(history[cur_hist])

def inputLineKeyReleased(code):
    pass

inputLine.connect("returnPressed()", interactiveInput);

File: src/LorrisAnalyzer/DataWidgets/ScriptWidget/examples/terminal.js
// Make ScriptWidget work as terminal
var history = new Array("");
var cur_hist = 0;
var last_idx = -1;

function onDataChanged(data, dev, cmd, index) {
    if(index != last_idx)
    {
        appendTerm(data);
        last_idx = index;
    }
}

function onKeyPress(key) {
    sendData(key);
}

// Input line - right click to script -> show input line
function interactiveInput()
{
    sendData(inputLine.text);


    if(cur_hist != history.length-1)
        history.splice(cur_hist, 1);
    cur_hist = history.length;
    history[cur_hist-1] = inputLine.text;
    history.push("");

    inputLine.clear();
}

// code is Qt::Key (http://qt-project.org/doc/qt-4.8/qt.html#Key-enum)
function inputLineKeyPressed(code) {
    switch(code) {
    case 0x01000013: // Qt::Key_U
        if(cur_hist > 0) {
            --cur_hist;
            inputLine.setText(history[cur_hist]);
        }
        break;
    case 0x01000015: // Qt::Key_Down
        if(cur_hist < history.length-1) {
            ++cur_hist;
            inputLine.setText(history[cur_hist]);
        }
        break;
    }
}

function inputLineKeyReleased(code) {
    return;
}

inputLine.returnPressed.connect(interactiveInput);
